frontend/src/logic/schedule/data-row.logic.ts

```diff
@@ -59,4 +63,37 @@ export class DataRow {
   public findValues(...args: string[]): string[] {
     return args.map((arg) => this.findValue(arg));
   }
+  //#endregion
+
+  //# row check region
+  private checkShiftRowPattern(): boolean {
+    const containsNotEmptyKey = this.data[0] != null && this.data[0] != "";
+
+    let shiftCodesList = Object.keys(ShiftCode)
+        .map(k => ShiftCode[k as any])
+    let containsAnyShiftCode = false;
+    shiftCodesList.forEach(shift => {
+      if(this.data.includes(shift)){
+        containsAnyShiftCode = true;
+      }
+    })
```

Comment:
To można było zapisać trochę krócej:
```
let containsShiftCode = this.data.filter( c => c in ShiftCode).length != 0
```
![image](https://user-images.githubusercontent.com/21079319/93493067-cc8d7300-f90b-11ea-82b3-056a350b4f2a.png)

Reply:
Tak coś czułem, że można to zrobić jakoś mądrzej. Zmienię to :)

Reply:
👍